cpp/src/ice2slice/Gen.cpp

```diff
@@ -299,28 +299,114 @@ namespace
         if (operation)
         {
             std::map<std::string, StringList> parameterDocs = comment->parameters();
-            ParameterList parameters = operation->parameters();
+
+            // Input parameters
+            ParameterList parameters = operation->inParameters();
             for (ParameterList::const_iterator it = parameters.begin(); it != parameters.end(); it++)
             {
                 ParameterPtr param = *it;
-                if (!param->isOutParam())
+                std::map<std::string, StringList>::const_iterator q = parameterDocs.find(param->name());
+                if (q != parameterDocs.end())
                 {
-                    std::map<std::string, StringList>::const_iterator q = parameterDocs.find(param->name());
-                    if (q != parameterDocs.end())
+                    out << nl << "/// @param " << param->name() << ": ";
+                    for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
                     {
-                        out << nl << "/// @param " << param->name() << ": ";
-                        for (StringList::const_iterator r = q->second.begin(); r != q->second.end();)
+                        if (r != q->second.begin())
                         {
-                            if (r != q->second.begin())
-                            {
-                                out << nl;
-                            }
-                            out << (*r);
-                            r++;
+                            out << nl;
+                            out << "/// ";
                         }
+                        out << (*r);
+                        r++;
                     }
                 }
             }
+
+            // Output parameters
+            parameters = operation->outParameters();
+            for (ParameterList::const_iterator it = parameters.begin(); it != parameters.end(); it++)
+            {
+                ParameterPtr param = *it;
+                std::map<std::string, StringList>::const_iterator q = parameterDocs.find(param->name());
+                if (q != parameterDocs.end())
+                {
+                    out << nl << "/// @returns " << param->name() << ": ";
```

Comment:
This does not look correct in the situation where we have a single out and no return. It would be good to add some tests.